src/api/v2/transactions.rs

```diff
@@ -0,0 +1,89 @@
+use anyhow::{Context, Result};
+use async_trait::async_trait;
+use avail_subxt::{
+	api::{self, runtime_types::sp_core::bounded::bounded_vec::BoundedVec},
+	avail,
+	primitives::AvailExtrinsicParams,
+	AvailConfig,
+};
+use sp_core::H256;
+use subxt::{ext::sp_core::sr25519::Pair, tx::PairSigner};
+
+#[async_trait]
+pub trait Submit {
+	async fn submit(&self, data: Vec<u8>) -> Result<H256>;
+}
+
+#[derive(Clone)]
+pub struct RawSubmit {
+	pub node_client: avail::Client,
+	pub app_id: u32,
+}
+
+// NOTE: Replace this with avail::PairSigner after implenting required traits in avail-subxt
```

Comment:
I think `TODO` should be more relevant here

Reply:
Makes sense 👍